vta/hardware/chisel/src/test/scala/ComputeTests.scala

```diff
@@ -0,0 +1,913 @@
+  step(1)
+  poke(c.io.biases.readdata, ("h"+biases4(5).substring(32 * 0, 32 * 0 + 32)).U)
+  step(1)
+
+  expect(c.io.biases.address, ("h"+(dram_base+BigInt("180",16)).toString(16)).U)
+  poke(c.io.biases.readdata, ("h"+biases4(6).substring(32 * 3, 32 * 3 + 32)).U) // data 6
+  step(1)
+  poke(c.io.biases.readdata, ("h"+biases4(6).substring(32 * 2, 32 * 2 + 32)).U)
+  step(1)
+  poke(c.io.biases.readdata, ("h"+biases4(6).substring(32 * 1, 32 * 1 + 32)).U)
+  step(1)
+  poke(c.io.biases.readdata, ("h"+biases4(6).substring(32 * 0, 32 * 0 + 32)).U)
+  step(1)
+  poke(c.io.gemm_queue.data, insn12)
+  poke(c.io.gemm_queue.valid, 1.U)
+
+  expect(c.io.biases.address, ("h"+(dram_base+BigInt("1c0",16)).toString(16)).U)
+  poke(c.io.biases.readdata, ("h"+biases4(7).substring(32 * 3, 32 * 3 + 32)).U) // data 7
+  step(1)
+  poke(c.io.biases.readdata, ("h"+biases4(7).substring(32 * 2, 32 * 2 + 32)).U)
+  step(1)
+  poke(c.io.biases.readdata, ("h"+biases4(7).substring(32 * 1, 32 * 1 + 32)).U)
+  step(1)
+  poke(c.io.biases.readdata, ("h"+biases4(7).substring(32 * 0, 32 * 0 + 32)).U)
+  step(1)
+  step(1)
+  step(1)
+  step(1)
+
+  step(1)
+  step(1)
+  step(1)
+  poke(c.io.gemm_queue.data, 0.U)
+  poke(c.io.gemm_queue.valid, 0.U)
+  step(1)
+  step(1)
+  expect(c.io.out_mem.write, 1.U)
+  expect(c.io.out_mem.address, "h00".U)
+  expect(c.io.out_mem.writedata, "hf5f0f5faf7f6f1040dfe000800b0ff1e".U)
+  step(1)
+  expect(c.io.out_mem.write, 1.U)
+  expect(c.io.out_mem.address, "h10".U)
+  poke(c.io.out_mem.waitrequest, 1.U)
+  step(1)
+  expect(c.io.out_mem.address, "h10".U)
+  expect(c.io.out_mem.writedata, "h06f0f9f1fd04f10cff0cf1f104fa07f1".U)
+  poke(c.io.out_mem.waitrequest, 0.U)
+  step(1)
+  expect(c.io.out_mem.write, 1.U)
+  expect(c.io.out_mem.address, "h20".U)
+  expect(c.io.out_mem.writedata, "h0ef5f1f601fafc06f0f5f5f70f0df1f2".U)
+  step(1)
+  expect(c.io.out_mem.write, 1.U)
+  expect(c.io.out_mem.address, "h30".U)
+  expect(c.io.out_mem.writedata, "h030c04fe0c080b0cfd0afbf8f602fcf3".U)
+  step(1)
+  expect(c.io.out_mem.address, "h40".U)
+  expect(c.io.out_mem.writedata, "h080e080ff50c0a06f2f00001fa00fa07".U)
+  step(1)
+  poke(c.io.out_mem.waitrequest, 1.U)
+  step(1)
+  expect(c.io.out_mem.address, "h50".U)
+  expect(c.io.out_mem.writedata, "hf10803fd05f6f20f0cfaf40f0bfe0f0f".U)
+  poke(c.io.out_mem.waitrequest, 0.U)
+  step(1)
+  expect(c.io.out_mem.address, "h60".U)
+  expect(c.io.out_mem.writedata, "h03f50c0ff4060d0ef60bf50f08040dfb".U)
+  step(1)
+  // poke(c.io.out_mem.waitrequest, 1.U)
+  // step(1)
+  expect(c.io.out_mem.address, "h70".U)
+  poke(c.io.out_mem.waitrequest, 0.U)
+  step(1)
+
+  step(1)
+  expect(c.io.out_mem.write, 0.U)
+
+  step(1)
+  expect(c.io.g2s_dep_queue.valid, 1.U)
+  poke(c.io.g2s_dep_queue.ready, 1.U)
+  expect(c.io.g2s_dep_queue.valid, 1.U)
+  step(1)
+  poke(c.io.g2s_dep_queue.ready, 0.U)
+  step(1)
+  step(1)
+  step(1)
+  step(1)
+
+  }
+  
+  init_compute0()
+  test_compute0()
```

Comment:
these test cases at least appear to be very repetitious. My guess is that they can be parameterized (which, in doing so, would make the meaning of 1 vs 2 vs 99 more transparent).